rpc_test.go

```diff
@@ -148,12 +140,40 @@ func benchmarkEchoGRPC(b *testing.B, size int) {
 	)
 }
 
-func BenchmarkGRPC_1K(b *testing.B) {
-	benchmarkEchoGRPC(b, 1<<10)
+// Serve
+
+func listenAndServeGRPCServe(listener net.Listener, _ *tls.Config) error {
+	grpcServer := grpc.NewServer()
+	RegisterEchoServer(grpcServer, &echoServer{})
+	return grpcServer.Serve(listener)
+}
+
+func BenchmarkGRPCServe_1K(b *testing.B) {
+	benchmarkEchoGRPC(b, listenAndServeGRPCServe, 1<<10)
+}
+
+func BenchmarkGRPCServe_64K(b *testing.B) {
+	benchmarkEchoGRPC(b, listenAndServeGRPCServe, 64<<10)
+}
+
+// ServeHTTP
+
+func listenAndServeGRPCServeHTTP(listener net.Listener, tlsConfig *tls.Config) error {
+	grpcServer := grpc.NewServer()
+	RegisterEchoServer(grpcServer, &echoServer{})
+	srv := http.Server{
+		TLSConfig: tlsConfig,
+		Handler:   grpcServer,
+	}
+	return srv.Serve(listener)
```

Comment:
@petermattis are you sure the latter's lifetime is a single request? it's a single HTTP request, but that request is a long-lived one that spans many RPCs, AFAICT.